src/core/processing/qgsnativealgorithms.cpp

```diff
@@ -2618,37 +2618,62 @@ QgsRasterLayerUniqueValuesReportAlgorithm *QgsRasterLayerUniqueValuesReportAlgor
   return new QgsRasterLayerUniqueValuesReportAlgorithm();
 }
 
+bool QgsRasterLayerUniqueValuesReportAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *layer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT" ), context );
+  mInterface.reset( layer->dataProvider()->clone() );
```

Comment:
Just one little thing - better check that layer is not nullptr, and if it is, return false here before trying to clone the provider